packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import path from 'path'
+import glob from 'glob'
+
+/**
+ * @class OverridesResolverPlugin
+ *
+ *  This plugin provides the "Overrides" behavior of the Template Extensibility feature,
+ *  allowing third party implementations that depend on an npm module for the base implementation
+ *  and then overriding only specific files
+ */
+class OverridesResolverPlugin {
+    /**
+     *
+     * @param options
+     * @param {string} options.overridesDir path to application base
+     * @param {string[]} options.overrides paths to overrides
+     * @param {string} options.projectDir path to project directory
+     */
+    constructor(options) {
+        this.overridesDir = options.overridesDir || ''
+        this.extends = options.extends || []
+        this.projectDir = options.projectDir
+        this._allSearchDirs = [this.projectDir + this.overridesDir, ...this.extends]
+        this.pkg = require(path.resolve(this.projectDir, 'package.json'))
+        this.extendsHashMap = new Map()
+
+        const OVERRIDES_EXTENSIONS = '.+(js|jsx|ts|tsx|svg|jpg|jpeg)'
+        const globPattern = `${this.pkg?.ccExtensibility?.overridesDir?.replace(
+            /\//,
+            ''
+        )}/**/*${OVERRIDES_EXTENSIONS}`
+        const overridesFsRead = glob.sync(globPattern)
+
+        const overrideReplace = this.pkg?.ccExtensibility?.overridesDir + '/'
+
+        overridesFsRead.forEach((item) => {
+            const end = item.substring(item.lastIndexOf('/index'))
+            const [l, ...rest] = item.split(/(index|\.)/)
+            this.extendsHashMap.set(
+                l.replace(/\/$/, '')?.replace(overrideReplace.replace(/\//, ''), ''),
+                [end, rest]
+            )
+        })
+    }
+
+    /**
+     *
+     * @param requestPath
+     * @param dirs
+     */
+    findFileMap(requestPath, dirs) {
```

Comment:
```suggestion
    findFileFromMap(requestPath, dirs) {
```

`findFileMap` implies the return value is a Map (or object), but it returns an actual file path.